src/test/java/io/neonbee/test/helper/MultipartResponse.java

```diff
@@ -0,0 +1,164 @@
+            if (nextIndex < 0) {
+                // this can only happen on the final delimiter line
+                break;
+            }
+
+            int endOfLineIndex = body.indexOf('\n', startIndex);
+            String partString = body.substring(endOfLineIndex + 1, nextIndex - 1);
+            responseParts.add(parseAsPart(partString));
+            startIndex = nextIndex;
+        }
+
+        return new MultipartResponse(batchResponse, responseParts);
+    }
+
+    @VisibleForTesting
+    static String extractBoundary(HttpResponse<Buffer> batchResponse) {
+        String contentType = Optional.ofNullable(batchResponse.headers().get(CONTENT_TYPE)).orElseThrow(
+                () -> new IllegalArgumentException("batchResponse does not contain mandatory header " + CONTENT_TYPE));
+        int indexOfBoundary = contentType.indexOf("boundary");
+        if (indexOfBoundary < 0) {
+            throw new IllegalArgumentException("batchResponse header " + CONTENT_TYPE + " does not contain boundary!");
+        }
+        String[] boundaryKeyAndValue = contentType.substring(indexOfBoundary).split("=", 2);
+        if (boundaryKeyAndValue.length < 2) {
+            throw new IllegalArgumentException(
+                    "Boundary not properly specified in batchResponse header " + CONTENT_TYPE + "!");
+        }
+        return boundaryKeyAndValue[1];
+    }
+
+    /**
+     * Parses a string expected to be a part of an <a href=
+     * "http://docs.oasis-open.org/odata/odata/v4.01/odata-v4.01-part1-protocol.html#sec_MultipartBatchResponse">OData
+     * Multipart Batch Response</a>. The string MUST NOT include the boundary delimiter.
+     *
+     * @param value string value of a batch response part
+     * @return part information extracted from given value string
+     */
+    @VisibleForTesting
+    static Part parseAsPart(String value) {
+        Collector<Map.Entry<String, String>, ?, Map<String, String>> headerCollector =
+                Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue);
+        Function<String, Map.Entry<String, String>> headerMapper = header -> {
+            String[] keyAndValue = header.split(":", 2);
+            return Map.entry(keyAndValue[0].trim(), keyAndValue[1].replaceFirst("\\R{1,2}$", "").trim());
+        };
+
+        String[] sections = value.split("\\R{2,}", 3);
+
+        String responseHead = sections[1];
+        Integer statusCode =
+                responseHead.lines().findFirst().map(status -> Iterables.get(Splitter.on(' ').split(status), 1))
+                        .map(Integer::parseInt)
+                        .orElseThrow(() -> new IllegalArgumentException(
+                                "Status code extraction from response part failed!"));
+        Map<String, String> responseHeaders = responseHead.lines().skip(1).map(headerMapper).collect(headerCollector);
+
+        return new Part(statusCode, caseInsensitiveMultiMap().addAll(responseHeaders), Buffer.buffer(sections[2]));
```

Comment:
```suggestion
        List<String> sections = Splitter.on("\n\n").splitToList(value);
        String statusLine = sections.get(1).substring(0,  sections.get(1).indexOf('\n'));
        int statusCode = Integer.parseInt(statusLine.replaceAll("(.*)(\\d{3})(.*)", "$2"));
        String headerLine = sections.get(1).substring(statusLine.length() +1);
        Map<String, String> headers = Splitter.on("\n").withKeyValueSeparator(Splitter.onPattern(": |:")) .split(headerLine);

        return new Part(statusCode, caseInsensitiveMultiMap().addAll(headers), Buffer.buffer(sections.get(2)));
```

Reply:
Some nice improvements in readability.
Thanks for that.
I will only keep the original solution for splitting into sections as the solution using splitToList would only partially read bodies containing more than one consecutive line break.